resources/domain/request/index.ts

```diff
@@ -1,3 +1,31 @@
+import { SignTypedDataVersion } from '@metamask/eth-sig-util'
+import type { SignatureRequestType, TypedMessage } from '../../../main/accounts/types'
+
 export const isCancelableRequest = (status: string): Boolean => {
   return !['sent', 'sending', 'verifying', 'confirming', 'confirmed', 'error', 'declined'].includes(status)
 }
+
+const eip2612PermitTypes = [
+  { name: 'owner', type: 'address' },
+  { name: 'spender', type: 'address' },
+  { name: 'value', type: 'uint256' },
+  { name: 'nonce', type: 'uint256' },
+  { name: 'deadline', type: 'uint256' }
+]
+
+export const isEip2612Permit = ({ data }: TypedMessage<SignTypedDataVersion>) => {
+  if (!('types' in data)) return false
+  if (!('Permit' in data.types) || !Array.isArray(data.types.Permit)) return false
+  const { Permit } = data.types
+
+  return (
+    Permit.length === eip2612PermitTypes.length &&
+    eip2612PermitTypes.every(({ name, type }) =>
+      Boolean(Permit.find((item) => item.name === name && item.type === type))
+    )
+  )
+}
+
+export const isSignatureRequest = (requestType: string): requestType is SignatureRequestType => {
+  return ['sign', 'signTypedData', 'signErc20Permit'].includes(requestType)
```

Comment:
do we need this new type? isnt it just a version of signTypedData?

Reply:
gotcha, yeah I think this makes sense for now though we might want to think about cleaning up the way these objects look in the future to prevent all of this "is it this OR this OR this OR this" type stuff where possible